mucommander-core/src/main/java/com/mucommander/ui/dialog/debug/DebugConsoleDialog.java

```diff
@@ -85,31 +107,66 @@ public class DebugConsoleDialog extends FocusDialog implements ActionListener, I
         Container contentPane = getContentPane();
 
-        loggingEventsList = new JList<>();
-        // Autoscroll when dragged
-        loggingEventsList.setAutoscrolls(true);
-        loggingEventsList.setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
-        loggingEventsList.setCellRenderer(new DebugListCellRenderer());
+        rootNode = new DefaultMutableTreeNode("Root-hidden");
         refreshLogRecords();
 
-        JScrollPane scrollPane = new JScrollPane(loggingEventsList, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+        loggingEventsTree = new JTree(rootNode);
+        loggingEventsTree.setAutoscrolls(true); // Autoscroll when dragged
+        loggingEventsTree.setScrollsOnExpand(true);
+        loggingEventsTree.setShowsRootHandles(true);
+        loggingEventsTree.setRootVisible(false);
+        var renderer = new DebugTreeCellRenderer();
+        loggingEventsTree.setCellRenderer(renderer);
+        renderer.setLeafIcon(null);
+        renderer.setClosedIcon(null);
+        renderer.setOpenIcon(null);
+
+        JScrollPane scrollPane = new JScrollPane(loggingEventsTree,
+                JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
         contentPane.add(scrollPane, BorderLayout.CENTER);
 
         JPanel southPanel = new JPanel(new BorderLayout());
         southPanel.add(createComboPanel(), BorderLayout.WEST);
 
         JPanel buttonPanel = new JPanel(new FlowLayout());
 
+        autoRefreshCheckBox = new JCheckBox(Translator.get("debug_console_dialog.auto_refresh"));
+        autoRefreshCheckBox.addItemListener((e) -> {
+            if (e.getStateChange() == ItemEvent.SELECTED) {
+                refreshButton.setEnabled(false);
+                periodicUpdater = executorService.scheduleAtFixedRate(
+                        () -> SwingUtilities.invokeLater(this::refreshLogRecords),
+                        INITIAL_DELAY, REFRESH_PERIOD, TimeUnit.MILLISECONDS);
+            } else {
+                refreshButton.setEnabled(true);
+                if (periodicUpdater != null) {
+                    periodicUpdater.cancel(false);
+                    periodicUpdater = null;
+                }
+            }
+        });
+        buttonPanel.add(autoRefreshCheckBox);
+
         refreshButton = new JButton(Translator.get(new RefreshAction.Descriptor().getLabel()));
-        refreshButton.addActionListener(this);
+        refreshButton.addActionListener((e) -> refreshLogRecords());
         buttonPanel.add(refreshButton);
 
         closeButton = new JButton(Translator.get("close"));
-        closeButton.addActionListener(this);
+        closeButton.addActionListener((e) -> {
+            if (periodicUpdater != null) {
+                periodicUpdater.cancel(false);
+                periodicUpdater = null;
+            }
+            executorService.shutdown();
```

Comment:
as you set `JDialog.DISPOSE_ON_CLOSE` as the default close operation, wouldn't it be better to extract the lines above to the `dispose` method?